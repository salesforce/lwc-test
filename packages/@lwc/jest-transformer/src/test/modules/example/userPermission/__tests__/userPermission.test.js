/*
 * Copyright (c) 2018, salesforce.com, inc.
 * All rights reserved.
 * SPDX-License-Identifier: MIT
 * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
 */
import { createElement } from 'lwc';
import UserPermission from 'example/userPermission';

jest.mock(
    '@salesforce/userPermission/mocked',
    () => {
        return { default: 'value set in test' };
    },
    { virtual: true }
);

afterEach(() => {
    while (document.body.firstChild) {
        document.body.removeChild(document.body.firstChild);
    }
});

describe('example-user-permission', () => {
    it('default snapshot', () => {
        const element = createElement('example-user-permission', { is: UserPermission });
        document.body.appendChild(element);
        expect(element).toMatchSnapshot();
    });

    it('returns default permission value as import path', () => {
        const element = createElement('example-user-permission', { is: UserPermission });
        document.body.appendChild(element);
        const userPermissionValue = document.body
            .querySelector('example-user-permission')
            .shadowRoot.querySelector('.unmockedPermission').textContent;
        expect(userPermissionValue).toBe('unmocked');
    });

    it('returns value from mock defined in test file', () => {
        const element = createElement('example-user-permission', { is: UserPermission });
        document.body.appendChild(element);
        const userPermissionValue = document.body
            .querySelector('example-user-permission')
            .shadowRoot.querySelector('.mockedPermission').textContent;
        expect(userPermissionValue).toBe('value set in test');
    });
});
