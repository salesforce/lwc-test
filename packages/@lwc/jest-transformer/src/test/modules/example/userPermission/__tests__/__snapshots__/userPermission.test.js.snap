// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`example-user-permission default snapshot 1`] = `
<example-user-permission>
  #shadow-root(open)
    <p
      class="mockedPermission"
    >
      value set in test
    </p>
    <p
      class="unmockedPermission"
    >
      unmocked
    </p>
</example-user-permission>
`;
